packages/hub/tasks/notify-merchant-claim.ts

```diff
@@ -0,0 +1,42 @@
+import { inject } from '@cardstack/di';
+import CardpaySDKService from '../services/cardpay-sdk';
+import Web3HttpService from '../services/web3-http';
+import WorkerClient from '../services/worker-client';
+
+interface MerchantClaimSubscriptionEvent {
+  returnValues: {
+    merchantSafe: string;
+    amount: string;
+  };
+}
+
+export default class NotifyMerchantClaim {
+  cardpay: CardpaySDKService = inject('cardpay');
+  web3: Web3HttpService = inject('web3-http', { as: 'web3' });
+  workerClient: WorkerClient = inject('worker-client', { as: 'workerClient' });
+
+  async perform(payload: MerchantClaimSubscriptionEvent) {
+    let merchantSafeAddress = payload.returnValues.merchantSafe;
+    let amountInWei = payload.returnValues.amount;
+
+    let web3Instance = this.web3.getInstance();
+    let Safes = await this.cardpay.getSDK('Safes', web3Instance);
+    let { safe } = await Safes.viewSafe(merchantSafeAddress);
```

Comment:
let's not do this--its a very indirect way to get the information that you actually need